### DISCUSSION TITLE
WebUSB / WebAssembly support

### DISCUSSION THREAD

**Discussion by beriberikix** [kevinmehall/nusb#82] [Ideas]
Hello! Browsing the `probe-rs` source code brought me here. I'd like to suggest adding support for WebUSB support. This is so `probe-rs` could be used in the browser via WebAssembly (see https://github.com/probe-rs/probe-rs/discussions/838.)

Previously it was using a package that relied on libusb but it appears that `nusb` would make targeting the web much easier!

**Comment by kevinmehall** [comment]
Yes, it would be a lot easier to deal with WebUSB in pure Rust than going through libusb. Opened https://github.com/kevinmehall/nusb/issues/83.
